content/ui-patterns/overlays.mdx

```diff
@@ -0,0 +1,415 @@
+    />
+</ImageContainer>
+<Caption>An anchored overlay attached to its source.</Caption>
+
+Keep anchored overlays out of the viewport edges. They should be at least 16px away from the viewport boundaries. Users shouldn't be blocked from scrolling the page while an anchored overlay is open, nor should a scrolling event in the page close the overlay.
+
+Primer uses [`anchoredPosition`](https://github.com/primer/behaviors/blob/main/docs/anchored-position.md) to calculate the position of an overlay.
+
+<ImageContainer>
+  <img
+    width="508"
+    src="https://user-images.githubusercontent.com/293280/147167707-66da9301-fdfd-4810-9f62-41e4f6f3c23c.png"
+    alt=" "
+    style="background: none"
+  />
+</ImageContainer>
+<Caption>
+
+A visualization of available [`anchoredPosition`](https://github.com/primer/behaviors/blob/main/docs/anchored-position.md) options. Source elements (purple squares) and their overlays (white squares) with  `side` (`outside-top`, `outside-right`, `outside-bottom`, `outside-left`) and `align` (`start`, `center`, `end`) settings.
+
+</Caption>
+
+### Centered overlays ("dialogs")
+
+Centered overlays appear centered in the viewport, with a visible backdrop that dims the rest of the window for focus.
+
+Centered overlays should always have an overlay header with title close button. Prevent page scrolling when a centered overlay is present.
+
+Overlays shouldn't touch the viewport edges. Use a safe 24px margin to keep centered overlays visually distinct from the rest of the page, even on narrow viewports.
+
+Simple and small centered overlays can remain as-is on narrow viewports. As more elements are added to it, responsive variants should be considered.
+
+<ImageContainer>
+  <img
+    width="661"
+    src="https://user-images.githubusercontent.com/293280/147170315-80da186f-7de8-4632-893e-31c56dc1e65d.png"
+    alt=" "
+    style="background: none"
+  />
+</ImageContainer>
+<Caption>
+
+A centered dialog with a title and close button in the header, a scrollable content region, and a footer with buttons.
+
+</Caption>
+
+### Side sheets
+
+Side sheets slide from the right or left edges of the viewport. Side sheets are mainly used for global navigation drawers, but right sheets can also be considered for quick previews in full-width pages, or quick configuration panels.
+
+Side sheets take the entire height of the viewport, and should be used sparingly. Don't use side sheets to present forms or flows that should be a page instead.
+
+## Responsive variants
+
+### Bottom sheets (specific to narrow viewports)
+
+A bottom sheet is a variant supported on narrow viewports made to facilitate reachability. Overlays that appear anchored or centered in large screens may adapt to appear as a bottom sheet on small screens.
+
+A bottom sheet variant can be used when the overlay content is too overwhelming to appear anchored. A bottom sheet always dims the rest of the screen for focus. Tapping on the backdrop dismisses the overlay.
```

Comment:
```suggestion
A bottom sheet variant can be used when the overlay content is too overwhelming to appear anchored. A bottom sheet always dims the rest of the screen for focus. Tapping on the backdrop dismisses the overlay.
```

Wondering if there's a more clear word instead of "overwhelming" here?

Reply:
@yaili how about "busy"?

> A bottom sheet variant can be used when the overlay content is too busy to appear anchored.

Reply:
yes that sounds good